lang/wasm.cc

```diff
@@ -27,6 +27,19 @@ context::context(wasm::engine* engine_ptr, std::string name) : engine_ptr(engine
 
 static constexpr size_t WASM_PAGE_SIZE = 64 * 1024;
 
+static uint32_t get_abi(wasmtime::Instance& instance, wasmtime::Store& store, uint8_t* data) {
+    auto abi_export = instance.get(store, "_scylla_abi");
+    if (!abi_export) {
+        throw wasm::exception(format("ABI version export not found - please export `_scylla_abi` in the wasm module"));
+    }
+    wasmtime::Global* abi_global = std::get_if<wasmtime::Global>(&*abi_export);
+    if (!abi_global) {
+        throw wasm::exception(format("Exported object {} is not a global", "_scylla_abi"));
+    }
+    wasmtime::Val abi_val = abi_global->get(store);
+    return *(uint32_t*)(data + abi_val.i32());
```

Comment:
Use read_le() to avoid failing on architectures that don't support unaligned access, or are big-endian.

Reply:
Ok